src/metrics-gatherer.ts

```diff
@@ -50,20 +71,23 @@ export class MetricsGatherer {
 		this.descriptions = {};
 		this.existMap = {};
 		this.kinds = {};
+		this.internalErrorCount = 0;
 	}
 
 	// fetch the description for a metric
+	@errGuard
 	public describe(name: string, text: string, custom: CustomParams = {}) {
 		if (this.descriptions[name]) {
-			throw new MetricsGathererError(
-				`tried to describe metric "${name}" twice`,
+			this.err(
+				new MetricsGathererError(`tried to describe metric "${name}" twice`),
```

Comment:
Just an observation: elsewhere in this file, `throw new` was replaced with `this.err` followed by `return`. But here there's no `return` clause. Maybe intentional, maybe not.

Reply:
Good catch!